src/utils.js

```diff
@@ -1,11 +1,31 @@
+const { hasOwnProperty, setPrototypeOf } = Object;
+let { apply } = typeof Reflect !== 'undefined' && Reflect;
+
+if (!apply) {
+  apply = function(fun, thisValue, args) {
+    return fun.apply(thisValue, args);
+  };
+}
+
 /* Add properties to a lookup table */
 export function addToSet(set, array) {
+  if (setPrototypeOf) {
+    // Make 'in' and truthy checks like Boolean(set.constructor)
+    // independent of any properties defined on Object.prototype.
+    // Prevent prototype setters from intercepting set as a this value.
+    setPrototypeOf(set, null);
+  }
   let l = array.length;
   while (l--) {
-    if (typeof array[l] === 'string') {
-      array[l] = array[l].toLowerCase();
+    let element = array[l];
+    if (typeof element === 'string') {
+      const lcElement = element.toLowerCase();
+      if (lcElement !== element) {
+        array[l] = lcElement;
```

Comment:
@mikesamuel I believe this is problematic when passing the newly-frozen config presets into `array`: #314